### PR TITLE
V1.3 User Class (matchPerson works) 

### PR DESCRIPTION
**Pull request by bennchong**
- [x] matchPerson()

User class extends Person is created inside ModelManager

### CODE REVIEW

**Comment by bennchong** [comment]
Register button now works

File: src/main/java/seedu/address/security/User.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    /***
+     *
+     * Finds @param name inside database to match it to the current user
+     *
+     */
+    private void matchPerson(String name) {
+        list = model.getAddressBook().getPersonList();
+        //Loops through personlist to get matched name Person Class
+        list.forEach((Person) -> {
```
Comment (nianfei97) [inline src/main/java/seedu/address/security/User.java]:
```suggestion
        for (Person person : list) {
            if (...) {
                ...
            }
        }
```

Comment (nianfei97) [reply src/main/java/seedu/address/security/User.java]: This way you get to avoid the weird Java lambda function thing

File: src/main/java/seedu/address/ui/RegistrationWindow.java [hunk 2]
```diff
@@ -30,6 +28,7 @@
     private TextField addressTextField;
     @FXML
     private Label label;
+    private Integer registerFlag;
```
Comment (nianfei97) [inline src/main/java/seedu/address/ui/RegistrationWindow.java]: Same as above, probably better and clearer to use an enum.

Reference: https://nuscs2113-ay1819s1.github.io/website/se-book-adapted/chapters/javaTools.html#enums

File: src/main/java/seedu/address/security/Security.java [hunk 3]
```diff
@@ -12,6 +12,17 @@
...
+    /***
+     * Logs out
+     */
+    void logout();
+
+    /***
+     * Registers the user
+     * Returns 1 if successful, 2 if user exists, 3 if the fields are incomplete
+     */
+    int register(String username, String password, String email, String phone, String address);
```
[the remark this answers is not in the document]
Comment (bennchong) [reply src/main/java/seedu/address/security/Security.java]: I went to check it out, enums are constants? How would it be helpful

[the remark this answers is not in the document]
Comment (bennchong) [reply src/main/java/seedu/address/security/Security.java]: I went to check it out, enums are constants? How would it be helpful

[the remark this answers is not in the document]
Comment (nianfei97) [reply src/main/java/seedu/address/security/Security.java]: https://crunchify.com/why-and-for-what-should-i-use-enum-java-enum-examples/

tl;dr 

> If you use enums instead of integers (or String codes), you increase compile-time checking and avoid errors from passing in invalid constants, and you document which values are legal to use.

File: src/main/java/seedu/address/security/SecurityManager.java [hunk 4]
```diff
@@ -42,4 +49,26 @@ public void login(String username, String password) {
...
+    public void logout() {
+        this.isAuthenticated = false;
+        //TODO Do I clear the User since its logged out? I can just leave it there to be overwritten
+        raise(new LogoutEvent());
+    }
+
+    @Override
+    public int register(String username, String password, String email, String phone, String address) {
+        try {
+            logic.execute("add n/" + username + " e/" + email + " p/" + phone + " a/" + address);
```
[the remark this answers is not in the document]
Comment (bennchong) [reply src/main/java/seedu/address/security/SecurityManager.java]: I think it looks more cluttered that way, with PREFIX_NAME + username + " " + PREFIX_EMAIL + ...

File: src/main/java/seedu/address/security/User.java (hunk 1 above)
Comment (bennchong) [reply src/main/java/seedu/address/security/User.java]: Lambda functions are quite commonly used. But ok i guess if that works ill use that way
